### ISSUE TITLE
Feature request: Automatic overwriting of tweens

### ISSUE THREAD

**Issue by alexanderflink-zz** [Demigiant/dotween#109]
Hello!
One thing I really miss in the DOTween platform (which is otherwise fantastic) is automatic overwriting of tweens with the same target.

For example: If I start tweening a certain transform's x position with one tween, and then start tweening it again with another tween, I would like the second tween to automatically take control instead of them "fighting" about it.

This is how it's handled in the Greensock animation platform (Javascript) and I think it works great. It's optional also, so you can set the overwrite mode to different modes.

Excerpt from Greensock Documentation on overwrite:
```
overwrite : String (or integer) - Controls how (and if) other tweens of the same target are overwritten. There are several modes to choose from, but "auto" is the default (although you can change the default mode using theTweenLite.defaultOverwrite property):
"none" (0) (or false) - no overwriting will occur.
"all" (1) (or true) - immediately overwrites all existing tweens of the same target even if they haven't started yet or don't have conflicting properties.
"auto" (2) - when the tween renders for the first time, it will analyze tweens of the same target that are currently active/running and only overwrite individual tweening properties that overlap/conflict. Tweens that haven't begun yet are ignored. For example, if another active tween is found that is tweening 3 properties, only 1 of which it shares in common with the new tween, the other 2 properties will be left alone. Only the conflicting property gets overwritten/killed. This is the default mode and typically the most intuitive for developers.
"concurrent" (3) - when the tween renders for the first time, it kills only the active (in-progress) tweens of the same target regardless of whether or not they contain conflicting properties. Like a mix of "all" and "auto". Good for situations where you only want one tween controling the target at a time.
"allOnStart" (4) - Identical to "all" but waits to run the overwrite logic until the tween begins (after any delay). Kills tweens of the same target even if they don't contain conflicting properties or haven't started yet.
"preexisting" (5) - when the tween renders for the first time, it kills only the tweens of the same target that existed BEFORE this tween was created regardless of their scheduled start times. So, for example, if you create a tween with a delay of 10 and then a tween with a delay of 1 and then a tween with a delay of 2 (all of the same target), the 2nd tween would overwrite the first but not the second even though scheduling might seem to dictate otherwise. "preexisting" only cares about the order in which the instances were actually created. This can be useful when the order in which your code runs plays a critical role.
```

**Comment by Demigiant** [comment]
Hello!

I know Greensock's library very well (was a big fan years ago when using Flash, and I respect it a lot). But in the case of DOTween, overwriting is not present by design. Let me explain...

Overwriting requires Reflection (Greensock uses a lot of it, not only for overwriting, but Flash/JS is a different beast), and in the Unity context Reflection is bad and slows things down. Sadly, there is really no other way to do it (I studied the matter a lot). So, while in my previous tween engine (HOTween) I did implement overwrite, here I decided to use a string-less/lighter architecture and scrap overwriting. I know it requires more attention from users, but in the end I believe it's better :P

Cheers,
Daniele
